taichi/program/program.cpp

```diff
@@ -253,6 +254,20 @@ SNode *Program::get_snode_root(int tree_id) {
   return snode_trees_[tree_id]->root();
 }
 
+SparseMatrixBuilder *
+Program::create_sparse_matrix_builder(int n, int m, uint64 max_num_entries) {
+  TI_ASSERT(arch_is_cpu(config.arch));
+  sparse_matrices_builder.emplace_back(
+      std::make_unique<SparseMatrixBuilder>(n, m, max_num_entries));
+  return sparse_matrices_builder.back().get();
```

Comment:
I believe it is fine for C++ to return a `std::unique_ptr` via pybind11. Here's an existing example:

* https://github.com/taichi-dev/taichi/blob/9d062754e5f6fa6d7b3b162cf11d5181e6ad1d43/taichi/program/program.cpp#L615
* https://github.com/taichi-dev/taichi/blob/b1c6b1998ff6f24135bf603a285f0df7aabb8a21/taichi/python/export_lang.cpp#L243